images/server/install-packages.sh

```diff
@@ -10,20 +10,28 @@ need_curl() {
     dnf install --setopt=install_weak_deps=False -y /usr/bin/curl
 }
 
-get_custom_repo() {
-    url="$1"
-    fname="$(basename "$url")"
-    dest="/etc/yum.repos.d/${fname}"
-    while [ -e "$dest" ]; do
-        u="$(echo "${url}" | sha256sum | head -c12)"
-        if [ -z "$u" ]; then
-            echo "failed to uniquify repo file name"
-            exit 2
+get_custom_repos() {
+    if [[ -n $1 ]]; then
+        urls=$1
+        need_curl
+        if [[ $(wc -w <<< "$1") -eq 1 || $ceph_from_custom -ne 1 ]]; then
+            get_distro_ceph_repo
+            urls=${1%% *}
         fi
-        dest="/etc/yum.repos.d/${u}-${fname}"
-    done
-    need_curl
-    curl -L "$url" -o "$dest"
+        for url in $urls; do
+            fname="$(basename "$url")"
+            dest="/etc/yum.repos.d/${fname}"
+            while [ -e "$dest" ]; do
```

Comment:
If the first iteration of the loop fails (i.e. the file with the hash prefixed still exists), this will become an infinite loop because the generated name will be the same.

I think this could be converted to a simple `if`. If the name with the hash exists, then it has been downloaded from the same place, so the file we'll download is most likely the same.